madprops/serializers.py

```diff
@@ -52,6 +53,12 @@ def to_representation(self, value):
         return result_dict
 
     def to_internal_value(self, data):
+        try:
+            data = dict(data)
+        except ValueError:
+            data_type = type(data).__name__
+            raise ValidationError(
```

Comment:
While "dict" may make sense to us, it's not a particularly valuable error message in the context of some endpoint consumer that has submitted JSON to the endpoint.